test/numerical/TestLoop.cpp

```diff
@@ -135,9 +135,9 @@ bool isOMLoopTheSameAsNaiveImplFor(std::string moduleIR,
     return false;
   }
 
-  auto *yRefInitShape = new int64_t[1]{1};
+  int64_t yRefInitShape = 1;
```

Comment:
The space management is different for yRefInitShape. Do we rely on the alloca/free for yRefInitShape in the current code? Or can we keep `new` and add an explicit free for the vFinalRef?

Reply:
dynamic memory is not needed. The create method only read the shape info, and copy it in its own data structure. So there is no need to use dynamic memory. Note that I changed the pattern as requested by @gongsu832 below.